### PR TITLE
LPS-184386 | Test Automation Creation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-161638 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-182417](https://issues.liferay.com/browse/LPS-182417)  → [LPS-184386](https://issues.liferay.com/browse/LPS-184386)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase [hunk 1]
```diff
@@ -93,4 +93,96 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182417. No accessibility issue in web content with date-time field structure"
+	@priority = 3
+	test ViewNoAccessibilityIssuesInDateTimeField {
```
Comment (rafaellajordao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:98]: Hi Tim! 
This is my first poshi accessibility test, I followed the documentation to create it but I realized that all the tests that have this `AssertAccessible();` function are set to Ignore, did I do it right or should I add the @ignore too?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase [hunk 2]
```diff
@@ -93,4 +93,96 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182417. No accessibility issue in web content with date-time field structure"
+	@priority = 4
+	test ViewNoAccessibilityIssuesInDateTimeField {
```
Comment (rafaellajordao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:98]: Hi @Tim-Cao !
This is my first poshi accessibility test, I followed the documentation to create it but I realized that all the tests that have this AssertAccessible(); function are set to Ignore, did I do it right or should I add the @ignore too?

Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:98]: The difference between `AssertAccessible();` and `AssertElementAccessible` , the first one to scan the whole page, the second one is only to scan an element like a field.
This case only need to scan relevant field not full page. So try to use AssertElementAccessible. I'll let you know after I tested on my local

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 3]
```diff
@@ -646,6 +646,186 @@ definition {
...
+			PortletEntry.inputTitle(title = "Web Content Title");
+
+			DERenderer.inputDataInDateField(
+				fieldLabel = "Date and Time",
+				index = 1,
+				value = "08/06/2021 10:00 AM");
+
+			DERenderer.inputDataInTextField(
+				fieldLabel = "Text",
+				value = "Test Text");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:770]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 4]
```diff
@@ -646,6 +646,186 @@ definition {
...
+			NavItem.gotoWebContent();
+
+			WebContentNavigator.gotoAddWithStructureCP(structureName = "WC Structure Name");
+
+			PortletEntry.inputTitle(title = "Web Content Title");
+
+			DERenderer.inputDataInDateField(
+				fieldLabel = "Date and Time",
+				index = 1,
+				value = "08/06/2021 10:00 AM");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:676]: Use WebContent.addWithStructureCP

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/fragments/Fragment.path [hunk 5]
```diff
@@ -356,7 +356,7 @@
 
 <tr>
 	<td>CONTRIBUTED_FRAGMENT_TEXT</td>
-	<td>xpath=(//div[contains(@id,'fragment-')][.//*[contains(@class,'${fragmentName}')]])[${index}]//*[contains(@data-lfr-editable-id,'${id}')]</td>
+	<td>xpath=(//div[contains(@id,'fragment-')][.//*[contains(@class,'${fragmentName}')]])[${index}]//*[contains(@data-lfr-editable-id,'${id}')] | //*[contains(@class,'${fragmentName}')][${index}]//*[contains(@data-lfr-editable-id,'${id}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/fragments/Fragment.path:359]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 6]
```diff
@@ -646,6 +646,186 @@ definition {
...
+				siteName = "Test Site Name");
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Date");
+		}
+
+		task ("Then you can map the date and time field to an editable field and customize the date and time format") {
+			PageEditorMapping.gotoEditableFieldMapping(
+				fragmentName = "Date",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:698]: Use className for special editable field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 7]
```diff
@@ -646,6 +646,186 @@ definition {
...
+			Select(
+				key_fieldLabel = "Date Format",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Custom");
+
+			Type(
+				key_fieldLabel = "Custom Format",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "dd-MM-YYYY HH:mm a");
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:719]: Add step to trigger auto save

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 8]
```diff
@@ -646,6 +646,186 @@ definition {
...
+				structureDescription = "WC Structure Description",
+				structureName = "WC Structure Name");
+
+			DataEngine.addField(
+				fieldFieldLabel = "Date and Time",
+				fieldName = "Date and Time");
+
+			DataEngine.addField(
+				fieldFieldLabel = "Text",
+				fieldName = "Text");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:753]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 9]
```diff
@@ -646,6 +646,186 @@ definition {
...
+
+			WebContentStructures.addCP(
+				structureDescription = "WC Structure Description",
+				structureName = "WC Structure Name");
+
+			DataEngine.addField(
+				fieldFieldLabel = "Date and Time",
+				fieldName = "Date and Time");
+
+			WebContentStructures.saveCP(structureName = "WC Structure Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:665]: Invalid parameter

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase [hunk 10]
```diff
@@ -93,4 +93,96 @@ definition {
...
+		var structureName = StringUtil.randomString(8);
+
+		var structureName = StringUtil.toLowerCase(${structureName});
+		var webContentName = StringUtil.randomString(8);
+
+		var webContentName = StringUtil.toLowerCase(${webContentName});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:104]: No need to use random name for assets inside a site with random name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase [hunk 11]
```diff
@@ -93,4 +93,96 @@ definition {
...
+
+			ContentPagesNavigator.openEditContentPage(
+				pageName = "Content Page Name",
+				siteName = ${randomSiteName});
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Date");
+
+			AssertAccessible();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:153]: The case doesn't have this step

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase [hunk 12]
```diff
@@ -93,4 +93,96 @@ definition {
...
+			Type(
+				key_fieldLabel = "Custom Format",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "dd-MM-YYYY HH:mm a");
+
+			PageEditor.waitForAutoSave();
+
+			AssertAccessible();
+
+			PageEditor.publish();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:184]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase [hunk 13]
```diff
@@ -93,4 +93,96 @@ definition {
...
+
+			Type(
+				key_fieldLabel = "Custom Format",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "dd-MM-YYYY HH:mm a");
+
+			PageEditor.waitForAutoSave();
+
+			AssertAccessible();
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentAccessibility.testcase:183]: Add steps to scan mapping panel

**Comment by Tim-Cao** [comment]
See changes https://github.com/liferay-echo/liferay-portal/pull/12109
@rafaellajordao

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**